insights/parsers/rsyslog_conf.py

```diff
@@ -10,70 +10,78 @@
 
 Due to high parsing complexity, this parser presents a simple line-based
 view of the file that meets the needs of the current rules.
-
-Example:
-    >>> content = '''
-    ... :fromhost-ip, regex, "10.0.0.[0-9]" /tmp/my_syslog.log
-    ... $ModLoad imtcp
-    ... $InputTCPServerRun 10514"
-    ... '''.strip()
-    >>> from insights.tests import context_wrap
-    >>> rsl = RsyslogConf(context_wrap(content))
-    >>> len(rsl)
-    3
-    >>> len(list(rsl))
-    3
-    >>> any('imtcp' in n for n in rsl)
-    True
 """
-from .. import Parser, parser, get_active_lines
-
-import re
+from .. import Parser, parser, get_active_lines, LegacyItemAccess
 from insights.specs import Specs
 
 
 @parser(Specs.rsyslog_conf)
-class RsyslogConf(Parser):
+class RsyslogConf(Parser, LegacyItemAccess):
     """
-    Parses `/etc/rsyslog.conf` info simple lines.
+    Parses `/etc/rsyslog.conf` content.
 
     Skips lines that begin with hash ("#") or are only whitespace.
 
     Attributes:
         data (list): List of lines in the file that don't start
             with '#' and aren't whitespace.
-        config_items(dict): Configuration items opportunistically found in the
-            configuration file, with their values as given.
+
+    Example:
+        >>> type(rsysconf)
+        <class 'insights.parsers.rsyslog_conf.RsyslogConf'>
+        >>> len(rsysconf)
+        13
+        >>> rsysconf[2]
+        'authpriv.*                                              /var/log/secure'
     """
+    def __init__(self, *args, **kwargs):
+        super(RsyslogConf, self).__init__(*args, **kwargs)
 
     def parse_content(self, content):
-        self.data = get_active_lines(content)
+        self.data = []
+        brace_flag = False
+        parenthesis_flag = False
+        parenthesis_string = ""
+        brace_string = ""
 
         self.config_items = {}
-        # Config items are e.g. "$Word value #optional comment"
-        config_re = re.compile(r'^\s*\$(?P<name>\S+)\s+(?P<value>.*?)(?:\s+#.*)?$')
-        for line in self.data:
-            lstrip = line.strip()
-            match = config_re.match(lstrip)
-            if match:
-                self.config_items[match.group('name')] = match.group('value')
 
-    def config_val(self, item, default=None):
```

Comment:
This would break backward compatibility for this parser so you need to keep it in the parser.  It seems like it might be better to have two parsers here since that are different, and use the `IsRhel[6|7|8]` components to trigger which parser is executed.

Reply:
self.config_items is used to store configuration like "$ModLoad imtcp", however, former code can not handle multi lines. Per our discussion in https://github.com/RedHatInsights/insights-core/pull/1952, I combine the multi lines into one line and not parse it as key-value format, directly return content line by line. Also, this code can handle rsyslog.conf in RHEL6|7|8, no need to use IsRhel[6|7|8] components. I cretaed a issue to remove this function, https://github.com/RedHatInsights/insights-core/issues/2130